server/api.go

```diff
@@ -388,6 +401,37 @@ func MapRouteBySchema(server *Server, dataStore db.DB, s *schema.Schema) {
 	}
 }
 
+func waitForNewChanges(w http.ResponseWriter, r *http.Request, context middleware.Context) {
+	etag := calculateResponseEtag(context)
+	if etag == r.Header.Get("If-None-Match") {
```

Comment:
Perhaps using If-None-Match is not such a good idea here... See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/If-None-Match . Spec would force us to return 304, while we want to Long Poll on this ETag and eventually get next version. It would also interfere with caches. We need something like: "X-Long-Poll-If-Mach: xxxx".